File: FrontEndWebMusic/src/apis/api_recommendation.js
import axios from 'axios';

export const getRecommendations = async (userId) => {
    try {
        const response = await axios.get(`http://localhost:8080/api/recommendations?userId=${userId}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching recommendations:', error);
        throw error;
    }
};

File: BackEndWebMusicApi/src/main/java/com/btec/webmusic_api/api_recommendation.java
package com.btec.webmusic_api;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.List;
import java.util.Map;

@Component
public class api_recommendation {

    private final RestTemplate restTemplate;

    @Autowired
    public api_recommendation(RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }

    public List<Map<String, Object>> getRecommendations(String userId) {
        String url = "http://localhost:5001/recommend?user_id=" + userId;
        ResponseEntity<List> response = restTemplate.getForEntity(url, List.class);
        return response.getBody();
    }
}
